common/lib/xmodule/xmodule/contentstore/content.py

```diff
@@ -351,27 +367,30 @@ def generate_thumbnail(self, content, tempfile_path=None, dimensions=None):
                 else:
                     im = Image.open(tempfile_path)
 
-                # I've seen some exceptions from the PIL library when trying to save palletted
-                # PNG files to JPEG. Per the google-universe, they suggest converting to RGB first.
-                im = im.convert('RGB')
+                # We use the context processor here to avoid leaking the inner file descriptor
+                # of the Image object -- this way it gets closed after we're done with using it.
+                thumbnail_file = StringIO.StringIO()
+                with im:
```

Comment:
This wasn't what I was expecting, and I don't know if it is correct. I expected this sort of syntax, the same as what you did in the SVG case:

````
with Image.open('test.png') as im:
    im = im.convert('RGB')
````

Did you find a reference supporting this different pattern?